runner.py

```diff
@@ -431,11 +435,17 @@ def train(model, cam, labels, opt, args, light=None, sched=None):
     # dn/dx -> 0, hopefully smoothes out the local normals of the surface.
     # TODO does it matter to normalize the normal or not?
     if args.smooth_normals > 0:
-      # TODO maybe lower dimensionality of n?
-      delta_n = torch.autograd.grad(
-        inputs=pts, outputs=F.normalize(n,dim=-1), create_graph=True,
-        grad_outputs=torch.ones_like(n),
-      )[0]
+      if args.smooth_eps > 0:
+        # epsilon-perturbation implementation from unisurf
+        perturbation = F.normalize(torch.rand(3, device=device), dim=-1) * args.smooth_eps
```

Comment:
Just curious, will this have some bias because it's not uniform on `[-1,1]`, but `[0,1]`?